src/software/backend/input/replay_logging/replay_reader.cpp

```diff
@@ -0,0 +1,139 @@
+#include "replay_reader.h"
+
+#include <google/protobuf/util/delimited_message_util.h>
+
+#include <fstream>
+
+// unfortunately protobuf forces me to write caveman code
+extern "C"
+{
+#include <fcntl.h>
+#include <unistd.h>
+}
+
+namespace fs = std::experimental::filesystem;
+
+namespace
+{
```

Comment:
what's the point of anon namespace? can we just have this function be free

Reply:
there wasn't really a point either way, but this function is not used anywhere else